### DISCUSSION TITLE
Adding Query entity via API

### DISCUSSION THREAD

**Discussion by astoulov** [open-metadata/OpenMetadata#13246] [Q&A] [answered]
Hi team,

I'm trying to add a query via API. Te Query is used in my custom dashboard (SAP BOE). But I get an error that it doesn't exist.
![image](https://github.com/open-metadata/OpenMetadata/assets/145361143/003a4ed7-fc55-4ef9-9908-a22cb944617d)

The custom dashboard exists in the system:
![image](https://github.com/open-metadata/OpenMetadata/assets/145361143/304a2196-3e39-4e5d-b025-d9f9495e4357)

Is it possible to link query with custom dashboard ?

**Comment by ulixius9** [comment, marked as answer]
hey @astoulov, the `queryUsedIn` parameter takes list [entityReference ](https://github.com/open-metadata/OpenMetadata/blob/2986d616b718bd3109a899cadb14e873b6073538/openmetadata-spec/src/main/resources/json/schema/type/entityReference.json#L13), where you need to specify the type as type of entity and not as the service type, so in your example the type should be `dashboard` and not the service type `CustomDashboard`
